// ng2-handy-syntax-highlighter/esm2015/ng2-handy-syntax-highlighter.js
import { ElementRef, Directive, Input, Component, NgModule } from '@angular/core';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class HighlighterDirective {
    /**
     * @param {?} eltRef
     */
    constructor(eltRef) {
        this.eltRef = eltRef;
    }
    /**
     * @return {?}
     */
    ngAfterViewInit() {
        this.eltRef.nativeElement.innerHTML = Prism.highlight(this.content, Prism.languages[this.language]);
    }
}
HighlighterDirective.decorators = [
    { type: Directive, args: [{
                selector: '[highlighter]'
            },] },
];
/** @nocollapse */
HighlighterDirective.ctorParameters = () => [
    { type: ElementRef, },
];
HighlighterDirective.propDecorators = {
    "language": [{ type: Input },],
    "content": [{ type: Input },],
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class HandySyntaxHighlighterComponent {
}
HandySyntaxHighlighterComponent.decorators = [
    { type: Component, args: [{
                selector: 'handy-syntax-highlighter',
                styles: [`.tag{font-size:14px!important;padding:0!important}.h1-top a{color:#1f2021}pre{font-size:14px!important;line-height:1.4!important;padding-bottom:0!important}`],
                template: `
        <pre class="language-{{language}}">
      <code highlighter [language]="language" [content]="content"></code>
    </pre>
    `
            },] },
];
/** @nocollapse */
HandySyntaxHighlighterComponent.ctorParameters = () => [];
HandySyntaxHighlighterComponent.propDecorators = {
    "language": [{ type: Input },],
    "content": [{ type: Input },],
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class Ng2HandySyntaxHighlighterModule {
}
Ng2HandySyntaxHighlighterModule.decorators = [
    { type: NgModule, args: [{
                declarations: [HighlighterDirective, HandySyntaxHighlighterComponent],
                exports: [HandySyntaxHighlighterComponent]
            },] },
];
/** @nocollapse */
Ng2HandySyntaxHighlighterModule.ctorParameters = () => [];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/**
 * Generated bundle index. Do not edit.
 */

export { Ng2HandySyntaxHighlighterModule, HandySyntaxHighlighterComponent as ɵb, HighlighterDirective as ɵa };
