// ng2-handy-syntax-highlighter/bundles/ng2-handy-syntax-highlighter.umd.js
(function (global, factory) {
	typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core')) :
	typeof define === 'function' && define.amd ? define('ng2-handy-syntax-highlighter', ['exports', '@angular/core'], factory) :
	(factory((global['ng2-handy-syntax-highlighter'] = {}),global.ng.core));
}(this, (function (exports,core) { 'use strict';

var HighlighterDirective = /** @class */ (function () {
    function HighlighterDirective(eltRef) {
        this.eltRef = eltRef;
    }
    HighlighterDirective.prototype.ngAfterViewInit = function () {
        this.eltRef.nativeElement.innerHTML = Prism.highlight(this.content, Prism.languages[this.language]);
    };
    return HighlighterDirective;
}());
HighlighterDirective.decorators = [
    { type: core.Directive, args: [{
                selector: '[highlighter]'
            },] },
];
HighlighterDirective.ctorParameters = function () { return [
    { type: core.ElementRef, },
]; };
HighlighterDirective.propDecorators = {
    "language": [{ type: core.Input },],
    "content": [{ type: core.Input },],
};
var HandySyntaxHighlighterComponent = /** @class */ (function () {
    function HandySyntaxHighlighterComponent() {
    }
    return HandySyntaxHighlighterComponent;
}());
HandySyntaxHighlighterComponent.decorators = [
    { type: core.Component, args: [{
                selector: 'handy-syntax-highlighter',
                styles: [".tag{font-size:14px!important;padding:0!important}.h1-top a{color:#1f2021}pre{font-size:14px!important;line-height:1.4!important;padding-bottom:0!important}"],
                template: "\n        <pre class=\"language-{{language}}\">\n      <code highlighter [language]=\"language\" [content]=\"content\"></code>\n    </pre>\n    "
            },] },
];
HandySyntaxHighlighterComponent.ctorParameters = function () { return []; };
HandySyntaxHighlighterComponent.propDecorators = {
    "language": [{ type: core.Input },],
    "content": [{ type: core.Input },],
};
var Ng2HandySyntaxHighlighterModule = /** @class */ (function () {
    function Ng2HandySyntaxHighlighterModule() {
    }
    return Ng2HandySyntaxHighlighterModule;
}());
Ng2HandySyntaxHighlighterModule.decorators = [
    { type: core.NgModule, args: [{
                declarations: [HighlighterDirective, HandySyntaxHighlighterComponent],
                exports: [HandySyntaxHighlighterComponent]
            },] },
];
Ng2HandySyntaxHighlighterModule.ctorParameters = function () { return []; };

exports.Ng2HandySyntaxHighlighterModule = Ng2HandySyntaxHighlighterModule;
exports.ɵb = HandySyntaxHighlighterComponent;
exports.ɵa = HighlighterDirective;

Object.defineProperty(exports, '__esModule', { value: true });

})));
