// ng2-handy-syntax-highlighter/esm5/ng2-handy-syntax-highlighter.js
import { ElementRef, Directive, Input, Component, NgModule } from '@angular/core';

var HighlighterDirective = /** @class */ (function () {
    function HighlighterDirective(eltRef) {
        this.eltRef = eltRef;
    }
    HighlighterDirective.prototype.ngAfterViewInit = function () {
        this.eltRef.nativeElement.innerHTML = Prism.highlight(this.content, Prism.languages[this.language]);
    };
    return HighlighterDirective;
}());
HighlighterDirective.decorators = [
    { type: Directive, args: [{
                selector: '[highlighter]'
            },] },
];
HighlighterDirective.ctorParameters = function () { return [
    { type: ElementRef, },
]; };
HighlighterDirective.propDecorators = {
    "language": [{ type: Input },],
    "content": [{ type: Input },],
};
var HandySyntaxHighlighterComponent = /** @class */ (function () {
    function HandySyntaxHighlighterComponent() {
    }
    return HandySyntaxHighlighterComponent;
}());
HandySyntaxHighlighterComponent.decorators = [
    { type: Component, args: [{
                selector: 'handy-syntax-highlighter',
                styles: [".tag{font-size:14px!important;padding:0!important}.h1-top a{color:#1f2021}pre{font-size:14px!important;line-height:1.4!important;padding-bottom:0!important}"],
                template: "\n        <pre class=\"language-{{language}}\">\n      <code highlighter [language]=\"language\" [content]=\"content\"></code>\n    </pre>\n    "
            },] },
];
HandySyntaxHighlighterComponent.ctorParameters = function () { return []; };
HandySyntaxHighlighterComponent.propDecorators = {
    "language": [{ type: Input },],
    "content": [{ type: Input },],
};
var Ng2HandySyntaxHighlighterModule = /** @class */ (function () {
    function Ng2HandySyntaxHighlighterModule() {
    }
    return Ng2HandySyntaxHighlighterModule;
}());
Ng2HandySyntaxHighlighterModule.decorators = [
    { type: NgModule, args: [{
                declarations: [HighlighterDirective, HandySyntaxHighlighterComponent],
                exports: [HandySyntaxHighlighterComponent]
            },] },
];
Ng2HandySyntaxHighlighterModule.ctorParameters = function () { return []; };

export { Ng2HandySyntaxHighlighterModule, HandySyntaxHighlighterComponent as ɵb, HighlighterDirective as ɵa };
